### DISCUSSION TITLE
Can I apply multiple masks to a layer? Or apply a mask to another mask?

### DISCUSSION THREAD

**Discussion by lianzhao** [visgl/deck.gl#7810] [Q&A]
<img width="774" alt="image" src="https://user-images.githubusercontent.com/536786/229274449-c72743b0-ecc6-451d-8e7b-0c219b118fc9.png">

I've tried the following code, with no luck.
```js
const mask1 = new PolygonLayer({
  id: "mask1",
  operation: "mask",
  // ...
});

const mask2 = new PolygonLayer({
  id: "mask2",
  operation: "mask",
  maskId: "mask1", // try to apply mask1 to this mask
  extensions: [new MaskExtension()],
  // ...
});

const layer = new PolygonLayer({
  maskId: "mask2",
  extensions: [new MaskExtension()],
})
```

**Comment by grahambates-code** [comment]
maybe you can use turf.difference to generate a single mask

**Reply by lianzhao** [reply]
Yes I can, but I'd rather not... As I need to update both masks' radius on every onAfterRender callback. I do not want to calculate the difference on CPU while animating.

**Reply by lianzhao** [reply]
BTW, the lineWidthUnits of mask2 is `pixels`, not `meters`😂

**Reply by grahambates-code** [reply]
Depends how many data points you are rendering I guess and user specs. I have used turf to calculate something on each frame and its fast enough to not to be noticeable on a modern machine
